WordPress/Jetpack/Classes/NUX/New Landing Screen/Views/InfiniteScrollerView.swift

```diff
@@ -0,0 +1,114 @@
+///
+/// By default user interaction with the scroll view is disabled, but it can be enabled if desired.
+///
+/// Currently the view only supports vertical scrolling, but could easily be updated to support horizontal.
+/// This class also currently makes the assumption that the view generated by the `viewBuilder` closure will be taller
+/// than its frame.
+class InfiniteScrollerView: UIScrollView {
+    /// Delegate to poll for the speed and direction of the scrolling offset. Polled at the default rate of `CADisplayLink`.
+    var scrollerDelegate: InfiniteScrollerViewDelegate?
+
+    private var displayLink: CADisplayLink?
+
+    /// Closure used to repeat a view to populate a `UIStackView` for scrolling.
+    /// Provides the index of the view in the `UIStackView`.
+    private var viewBuilder: ((Int) -> UIView)?
+    private var stackView: UIStackView?
+
+    /// Returns the height of the first view in the stack view, if one exists. Used to compute reset points for vertical scrolling.
+    private var singleViewHeight: CGFloat? {
+        guard let stackView = stackView, !stackView.arrangedSubviews.isEmpty else {
+            return nil
+        }
+
+        return stackView.arrangedSubviews[0].frame.size.height
+    }
+
+    /// Initializes a view that produces an infinite scrolling effect.
+    /// - Parameter viewBuilder: Closure that generates a `UIView`. Repeatedly called to fill a `UIStackView` that will be scrolled.
+    convenience init(_ viewBuilder: @escaping ((Int) -> UIView)) {
+        self.init()
+        self.viewBuilder = viewBuilder
+        displayLink = CADisplayLink(target: self, selector: #selector(step))
+
+        setupScrollView()
+        let stackView = setupStackView()
+        addSubview(stackView)
+
+        stackView.topAnchor.constraint(equalTo: topAnchor).isActive = true
+        stackView.bottomAnchor.constraint(equalTo: bottomAnchor).isActive = true
+        stackView.leadingAnchor.constraint(equalTo: leadingAnchor).isActive = true
+        stackView.trailingAnchor.constraint(equalTo: trailingAnchor).isActive = true
+        stackView.widthAnchor.constraint(equalTo: widthAnchor).isActive = true
+
+        self.stackView = stackView
+        addViewsToStackView()
+
+        displayLink?.add(to: .current, forMode: .default)
+    }
+
+    private func addViewsToStackView() {
+        guard let viewBuilder = viewBuilder else { return }
+        guard let stackView = stackView else { return }
+
+        /// Note: This class makes the assumption that the view generated by `viewBuilder()` is taller than
+        /// the scroll view's frame. To support views shorter than the frame it'd need to calculate how many times
+        /// to repeat it based on the built view's height.
+        ///
+        /// Currently, five views are generated because of an even / odd congruency requirement by the JetpackLandingScreen.
+        for i in 0...4 {
```

Comment:
it would be nice to isolate magic numbers, just in case we need to reconfigure something without the need to diving into the code for that.

Reply:
Yep, agreed. [Addressed here](https://github.com/wordpress-mobile/WordPress-iOS/pull/19410#discussion_r991684940) which should now make this view generic and compatible with all cases by just using three views.